nanoFramework.System.Net.Http/Http/System.Net._InputNetworkStreamWrapper.cs

```diff
@@ -200,7 +200,7 @@ public override int Read(byte[] buffer, int offset, int size)
             if (m_chunk.m_Size == 0)
             {
                 // Nothing to read and actually it is the end of the message body. It is "case 4".
-                return 0;
+                return -1;
```

Comment:
I'm afraid we can't go with this _very_ breaking change... 😯 
This has to return 0.

I was thinking that the fix it's elsewhere. When the stream content is created, doesn't the stream length get set to 0? If that's the case, than it's a matter of not entering the _content.Read loop above and "just" return.

Reply:
Hmm, I presumed the InputNetworkStreamWrapper is only used in context of processing StreamContent so it would not break anything.
Returning -1 was an easy way out to let StreamContent know everything is processed.

However, while going deeper into the code, I believe there is a bigger underlying issue. I'll add my thoughts to the issue to discuss further there.